### PR TITLE
Version 1.1.0

### PR DESCRIPTION
**Pull request by andrasguseo**
= [1.1.0] 2024-09-12 =

* Feature - Add the `tec_labs_wpai_is_post_type_set` filter to allow force importing when the post type is not defined in the source.
* Tweak - Add more details to some log messages.

### CODE REVIEW

**Review by sdokus — APPROVED** [review]

File: src/TEC/Import/Post_Handler.php [hunk 1]
```diff
@@ -131,9 +131,24 @@ public function maybe_create_post( bool $continue_import, array $data, int $impo
...
 
-			return false;
+			/**
+			 * A filter to allow forcing the import even if post type is not supported or not set.
+			 *
+			 * @since 1.1.0
+			 *
+			 * @param bool $continue Whether the import should be forced to continue. Default: false.
+			 */
+			$continue = apply_filters( 'tec_labs_wpai_is_post_type_set', false );
```
Comment (dpanta94) [inline src/TEC/Import/Post_Handler.php:143]: Why dont we filter inside $this->get_supported_post_types( false ) instead ?

File: src/TEC/Import/Post_Handler.php [hunk 2]
```diff
@@ -131,9 +131,24 @@ public function maybe_create_post( bool $continue_import, array $data, int $impo
...
+
+			$msg .= $continue ? '`tec_labs_wpai_is_post_type_set` override in place, post will be imported.' : 'Skipping.' ;
+			$this->add_to_log( $msg );
+
+			// Bail, if there is no override for the non-supported post type.
+			if ( ! $continue ) {
+				return false;
+			}
```
Comment (dpanta94) [inline src/TEC/Import/Post_Handler.php:151]:
```suggestion
		return $continue;
```

You can return early with the `$continue` varialbe itself. Continueing with a posttype not supported or empty requires more changes :D

**Review by dpanta94 — CHANGES_REQUESTED** [review]

**Review by dpanta94 — APPROVED** [review]

File: src/TEC/Import/Post_Handler.php (hunk 2 above)
Comment (andrasguseo) [reply src/TEC/Import/Post_Handler.php:151]: I guess with the new filter we can disregard this?

**Review by dpanta94 — APPROVED** [review]

File: src/TEC/Import/Post_Handler.php (hunk 2 above)
Comment (dpanta94) [reply src/TEC/Import/Post_Handler.php:151]: Yes!

**Pull request merged**